core/src/main/scala/kafka/log/Log.scala

```diff
@@ -392,18 +392,27 @@ class Log(@volatile var dir: File,
         // if it's a log file, load the corresponding log segment
         val baseOffset = offsetFromFile(file)
         val timeIndexFileNewlyCreated = !Log.timeIndexFile(dir, baseOffset).exists()
+        val offsetIndexExists = Log.offsetIndexFile(dir, baseOffset).exists()
         val segment = LogSegment.open(dir = dir,
           baseOffset = baseOffset,
           config,
           time = time,
           fileAlreadyExists = true)
-
-        try segment.sanityCheck(timeIndexFileNewlyCreated)
-        catch {
-          case _: NoSuchFileException =>
+        
+        try {
+          // Resize the time index file to 0 if it is newly created.
+          if (timeIndexFileNewlyCreated)
+            segment.timeIndex.resize(0)
+          // Rebuild the index if the index file does not exist
+          if (!offsetIndexExists) {
```

Comment:
If we want to recover segment when offset index does not exist (which I think is a safe choice), maybe we should also recover segment when time index does not exist for the same reason and for consistency.

Reply:
Done.